src/plugins/dam/components/batch-edit-modal/BatchEdit.jsx

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import swal from 'sweetalert2';
+import { connect } from 'react-redux';
+
+import createDelegateFactory from '@triniti/app/createDelegateFactory';
+import {
+  Button,
+  Card,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+} from '@triniti/admin-ui-plugin/components';
+import NodeRef from '@gdbots/schemas/gdbots/ncr/NodeRef';
+
+import delegateFactory, { Delegate } from './delegate';
+import Form from './Form';
+import selector from './selector';
+
+import './styles.scss';
+
+
+class BatchEdit extends React.Component {
+  static async confirmDone(text) {
+    return swal.fire({
+      cancelButtonClass: 'btn btn-secondary',
+      confirmButtonClass: 'btn btn-danger',
+      confirmButtonText: 'Yes',
+      showCancelButton: true,
+      text,
+      title: 'Are you sure?',
+      type: 'warning',
+      reverseButtons: true,
+    });
+  }
+
+  static propTypes = {
+    assetIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+    allowMultiUpload: PropTypes.bool,
+    currentValues: PropTypes.shape({
+      credit: PropTypes.object,
+      expiresAt: PropTypes.instanceOf(Date),
+    }),
+    delegate: PropTypes.instanceOf(Delegate).isRequired,
+    getNode: PropTypes.func,
+    hasMultipleFiles: PropTypes.bool,
+    initialValues: PropTypes.shape({}),
+    isFormDirty: PropTypes.bool,
+    isFormPrestine: PropTypes.bool,
+    isFormValid: PropTypes.bool,
+    isOpen: PropTypes.bool,
+    nodeRef: PropTypes.instanceOf(NodeRef).isRequired,
+    /* eslint react/no-unused-prop-types: off */
+    onClose: PropTypes.func, // This is used in the delegate
+    onToggleBatchEdit: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    allowMultiUpload: true,
+    currentValues: {},
+    hasMultipleFiles: false,
+    getNode: noop,
+    initialValues: {},
+    isFormDirty: undefined,
+    isFormPrestine: undefined,
+    isFormValid: undefined,
+    isOpen: false,
+    onClose: noop,
+  };
+
+  constructor(props) {
+    super(props);
+    const {
+      delegate,
+    } = props;
+
+    delegate.bindToComponent(this);
+
+    this.handleToggleUploader = this.handleToggleUploader.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+  }
+
+  handleToggleUploader() {
+    const {
+      isFormDirty,
+      onToggleBatchEdit,
+    } = this.props;
+
+
+    if (isFormDirty) {
+      const confirmText = 'Do you want to leave the form without saving?';
+      BatchEdit.confirmDone(confirmText).then(({ value }) => {
+        if (value) {
+          onToggleBatchEdit();
+        } else {
+          // do nothing, user declined
+        }
+      });
+    } else {
+      onToggleBatchEdit();
+    }
+  }
+
+  handleUpdate() {
+    const {
+      assetIds,
+      currentValues,
+      delegate,
+      onToggleBatchEdit,
+    } = this.props;
+
+    // Slug is in currentValues sometimes so we must have a white list.
+    const filter = ['title', 'credit', 'expiresAt', 'description'];
```

Comment:
@rsumilang it sometimes has slug because of the sluggable subscriber's onTitleChanged. you could check there to see if the form even has a slug field registered to avoid this workaround. the subscriber's method shouldnt really be adding a slug value for this anyways